In [1]:
import pandas as pd
import numpy as np
from scipy import stats 

In [4]:
aqi = pd.read_csv('C:\\Users\\hp\\Desktop\\PYTHON\\Stat\\Day 1\\c4_epa_air_quality.csv')
aqi.head()

Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

In [6]:
print('Use head() to show a sample of data')
print(aqi.head())

Use head() to show a sample of data
   Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1 

In [8]:
print('Use describe() to summarize AQI')
print(aqi.describe())

Use describe() to summarize AQI
       Unnamed: 0  arithmetic_mean         aqi
count  260.000000       260.000000  260.000000
mean   129.500000         0.403169    6.757692
std     75.199734         0.317902    7.061707
min      0.000000         0.000000    0.000000
25%     64.750000         0.200000    2.000000
50%    129.500000         0.276315    5.000000
75%    194.250000         0.516009    9.000000
max    259.000000         1.921053   50.000000


In [11]:
print("For a more thorough examination of observations by state use values_counts()")
print(aqi['state_name'].value_counts())

For a more thorough examination of observations by state use values_counts()
state_name
California              66
Arizona                 14
Ohio                    12
Florida                 12
Texas                   10
New York                10
Pennsylvania            10
Michigan                 9
Colorado                 9
Minnesota                7
New Jersey               6
Indiana                  5
North Carolina           4
Massachusetts            4
Maryland                 4
Oklahoma                 4
Virginia                 4
Nevada                   4
Connecticut              4
Kentucky                 3
Missouri                 3
Wyoming                  3
Iowa                     3
Hawaii                   3
Utah                     3
Vermont                  3
Illinois                 3
New Hampshire            2
District Of Columbia     2
New Mexico               2
Montana                  2
Oregon                   2
Alaska                   2
Georgia              

Question 1: From preceding data exploration, what do you recognize?
• You have county-level data for the first hypothesis.
• Ohio and New York both have a higher number of observations to work with in this dataset.

Statistical Tests

1. Formulate the null hypothesis and the alternative hypothesis.
2. Set the significance level.
3. Determine the appropriate test procedure.
4. Compute the p-value.
5. Draw your conclusion.

In [21]:
# Create dataframes for each sample being compared in your test

ca_la = aqi[aqi['county_name']=='Los Angeles']
ca_other = aqi[(aqi['state_name']=='California') & (aqi['county_name']!='Los Angeles')]

In [22]:
 # For this analysis, the significance level is 5%
significance_level = 0.05

In [23]:
# Compute your p-value here

stats.ttest_ind(a=ca_la['aqi'], b=ca_other['aqi'], equal_var=False)

TtestResult(statistic=2.1107010796372014, pvalue=0.049839056842410995, df=17.08246830361151)

With a p-value (0.049) being less than 0.05 (as your significance level is
5%), reject the null hypothesis in favor of the alternative hypothesis.
Therefore, a metropolitan strategy may make sense in this case.

Hypothesis 2: With limited resources, ROA has to choose between New York
and Ohio for their next regional office. Does New York have a lower AQI than
Ohio?


In [24]:
# Create dataframes for each sample being compared in your test

ny = aqi[aqi['state_name']=='New York']
ohio = aqi[aqi['state_name']=='Ohio']

In [25]:
# Compute your p-value
tstat, pvalue = stats.ttest_ind(a=ny['aqi'], b=ohio['aqi'], alternative='less',equal_var=False)
print(tstat)
print(pvalue)


-2.025951038880333
0.03044650269193468


With a p-value (0.030) of less than 0.05 (as your significance level is
5%) and a t-statistic < 0 (-2.036), reject the null hypothesis in favor of the alternative
hypothesis.

Therefore, you can conclude at the 5% significance level that New York has a lower mean AQI than
Ohio.

Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or
greater. Will Michigan be affected by this new policy?


In [26]:
# Create dataframes for each sample being compared in your test

michigan = aqi[aqi['state_name']=='Michigan']

In [28]:
# Compute your p-value
tstat, pvalue = stats.ttest_1samp(michigan['aqi'], 10, alternative='greater')
print(tstat)
print(pvalue)

-1.7395913343286131
0.9399405193140109


With a p-value (0.939) being greater than 0.05 (as your significance level
is 5%) and a t-statistic < 0 (-1.74), fail to reject the null hypothesis.
Therefore, you cannot conclude at the 5% significance level that Michigan’s mean AQI is greater
than 10. This implies that Michigan would most likely not be affected by the new policy.

Question 5. Did your results show that the AQI in Los Angeles County was statistically
different from the rest of California? Yes, the results indicated that the AQI in Los Angeles
County was in fact different from the rest of California.

Question 6. Did New York or Ohio have a lower AQI? Using a 5% significance level, you
can conclude that New York has a lower AQI than Ohio based on the results.

Question 7: Will Michigan be affected by the new policy impacting states with a mean
AQI of 10 or greater? Based on the tests, you would fail to reject the null hypothesis, meaning
you can’t conclude that the mean AQI is greater than 10. Thus, it is unlikely that Michigan would
be affected by the new policy.
